resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -687,6 +687,20 @@ private[spark] class Client(
   private def createConfArchive(): File = {
     val hadoopConfFiles = new HashMap[String, File]()
 
+    // SPARK_CONF_DIR shows up in the classpath before HADOOP_CONF_DIR/YARN_CONF_DIR
+    val localConfDir = System.getProperty("SPARK_CONF_DIR",
+      System.getProperty("SPARK_HOME") + File.separator + "conf")
+    val dir = new File(localConfDir)
+    if (dir.isDirectory) {
+      val files = dir.listFiles(new FileFilter {
+        override def accept(pathname: File): Boolean = {
+          pathname.isFile && pathname.getName.endsWith("xml")
+        }
+      })
+      files.foreach { f => hadoopConfFiles(f.getName) = f }
+    }
+
+    // Ensure HADOOP_CONF_DIR/YARN_CONF_DIR not overriding existing files
```

Comment:
This comment doesn't make a lot of sense, at least not in this position. What are you trying to say?

Reply:
ok, i'd remove it